### PR TITLE
Pr add recursion limit check error code

### PR DESCRIPTION
**Pull request by elcritch**
Same feature as in https://github.com/nim-lang/sat/pull/2 but using a result code.

### CODE REVIEW

**Comment by elcritch** [comment]
Integrated switch to unit-test PR too.

**Comment by Araq** [comment]
Much more elegant:

```nim

proc satisfiableIter(f: Formular; sout: var Solution; iters: var int): bool =
  if iters == 0:
    return false
  dec iters
  ...

```

And then the caller can just check for the value of `iters` after the recursive shenanigans and introduce the 3 value logic.

**Comment by elcritch** [comment]
> And then the caller can just check for the value of `iters` after the recursive shenanigans and introduce the 3 value logic.

Nice, I like it. The 3-value logic is annoying.

**Comment by elcritch** [comment]
I went with this for the public api. Mainly because I find having to create a `var iters` to call the api every time annoying. Generally this should only occur in degenerate cases anyway:

```nim
proc satisfiable*(f: Formular; sout: var Solution;
                   maxIters: int = MaxDefaultIterations): bool {.raises: [SatOverflowError].} =
   ## Determines if the SAT problem given in the given Formular `f` is satisfiable.
   var iters = maxIters
   result = satisfiableIter(f, sout, iters)
   if iters == 0:
     raise newException(SatOverflowError, "SAT iteration limit exceeded")
```

File: README.md [hunk 1]
```diff
@@ -3,4 +3,8 @@ A SAT solver written in Nim.
 
 ```nim
 import sat/[sat, satvars]
-```
\ No newline at end of file
+```
+
+Note:
```
Comment (Araq) [inline README.md:8]: Note: We really need to fix this one too.

Comment (elcritch) [reply README.md:8]: Yah, seems like a gotcha.

**Pull request merged**